File: set2/ex09/pkcs.rb
class Pkcs7
  attr_reader :plaintext
  def initialize(plaintext, padded_length)
    @plaintext = plaintext.bytes
    @padded_length = padded_length
  end

  def pad
    diff = @padded_length - @plaintext.length
    1.upto(diff) do
      @plaintext << diff
    end
  end
end

File: set2/ex09/pkcs.py
import binascii

class Pkcs7(object):
    def __init__(self, plaintext, padded_length):
        self.plaintext = bytearray(plaintext, 'ascii')
        self.padded_length = padded_length

    def pad(self):
        if self.padded_length < len(self.plaintext):
            raise ValueError("bad padded length for input plaintext")
        diff = self.padded_length - len(self.plaintext)
        while len(self.plaintext) < self.padded_length:
            self.plaintext.append(diff)


        
